validator/sawtooth_validator/journal/consensus/genesis/genesis_consensus.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ------------------------------------------------------------------------------
+from sawtooth_validator.journal.block_wrapper import NULL_BLOCK_IDENTIFIER
+from sawtooth_validator.journal.consensus.consensus import \
+    BlockPublisherInterface
+from sawtooth_validator.journal.consensus.consensus import \
+    BlockVerifierInterface
+from sawtooth_validator.journal.consensus.consensus import \
+    ForkResolverInterface
+
+
+class BlockPublisher(BlockPublisherInterface):
+    """The Genesis BlockPublisher is the basic publisher used only during the
+    production of a genesis block. This block is marked with consensus field of
+    `'Genesis'` and finalized as such.
+    """
+    def __init__(self,
+                 block_cache,
+                 state_view_factory,
+                 batch_publisher,
+                 data_dir):
+        super().__init__(block_cache,
+                         state_view_factory,
+                         batch_publisher,
+                         data_dir)
+
+    def initialize_block(self, block_header):
```

Comment:
I realize that the consensus factory is supposed to only return genesis as the consensus if it is the first block but as a safety precaution, maybe have `initialize_block`, `check_publish_block`, and `finalize_block` only return true if the block ID is the `NULL_BLOCK_IDENTIFIER`....or put asserts here to catch programmer errors in case of changes in the future.